src/main/java/vazkii/botania/common/item/equipment/bauble/ItemBauble.java

```diff
@@ -32,23 +36,41 @@
 import baubles.common.container.InventoryBaubles;
 import baubles.common.lib.PlayerHandler;
 import cpw.mods.fml.common.Optional;
+import cpw.mods.fml.common.eventhandler.SubscribeEvent;
+import cpw.mods.fml.common.gameevent.PlayerEvent;
+import cpw.mods.fml.common.network.FMLNetworkEvent;
 import cpw.mods.fml.relauncher.Side;
 import cpw.mods.fml.relauncher.SideOnly;
 
 @Optional.Interface(modid = "Thaumcraft", iface = "thaumcraft.api.IRunicArmor")
 public abstract class ItemBauble extends ItemMod implements IBauble, ICosmeticAttachable, IPhantomInkable, IRunicArmor {
 
-	private static final String TAG_HASHCODE = "playerHashcode";
-	private static final String TAG_HASHCODE_CLIENT = "playerHashcodeClient";
 	private static final String TAG_BAUBLE_UUID_MOST = "baubleUUIDMost";
 	private static final String TAG_BAUBLE_UUID_LEAST = "baubleUUIDLeast";
 	private static final String TAG_COSMETIC_ITEM = "cosmeticItem";
 	private static final String TAG_PHANTOM_INK = "phantomInk";
 
+	private static final HashMap<UUID, UUID> itemToPlayerRemote = new HashMap<>();
+	private static final HashMap<UUID, UUID> itemToPlayer = new HashMap<>();
+	private static final HashSet<UUID> toRemoveItems = new HashSet<>();
+
 	public ItemBauble(String name) {
 		super();
 		setMaxStackSize(1);
 		setUnlocalizedName(name);
+		MinecraftForge.EVENT_BUS.register(this);
```

Comment:
This will register a new event subscriber for each item that extends `ItemBauble`, but this seems to only handle static data. I would put these event handlers in a more general place that is only registered once